net/src/packet/mod.rs

```diff
@@ -202,43 +206,48 @@ impl<Buf: PacketBufferMut> Packet<Buf> {
 
         // build UDP header for Vxlan, setting ports, length and checksum.
         let mut udp = Udp::new(udp_src_port, Vxlan::PORT);
-        udp.set_checksum(0); /* optional in ipv4 (must be zero if unused) */
-
         #[allow(clippy::cast_possible_truncation)] // checked
         let udp_len = NonZero::new(len as u16).unwrap_or_else(|| unreachable!());
         #[allow(unsafe_code)] // sound usage due to length check
         unsafe {
             udp.set_length(udp_len);
         }
 
+        // the VXLAN spec says that the checksum SHOULD be zero
+        udp.set_checksum(UdpChecksum::ZERO);
```

Comment:
Really, I wonder why?  That seems busted, but oh well.

Reply:
https://datatracker.ietf.org/doc/html/rfc7348#section-5